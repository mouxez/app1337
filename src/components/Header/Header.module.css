.container {
    position: fixed;
    display: flex;
    justify-content: space-between;
    align-items: center;
    top: 0;
    width: calc(100% - 20px);
}

@media (max-width: 768px) {
    .container {
        flex-direction: column;
        justify-content: center;
    }
}

.title {
    font-weight: 600;
    font-size: 34px;
    font-family: 'Courier New', Courier, monospace;
    text-transform: uppercase;
    margin: 0;
}

.controlsWrap>button:nth-child(2) {
    margin-left: 10px;
}