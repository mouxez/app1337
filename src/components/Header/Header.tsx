import { Dispatch, FC, SetStateAction } from "react";
import { Radio, RadioChangeEvent } from "antd";
import Card from "../Card/Card";
import styles from "./Header.module.css";

const { container, title, controlsWrap } = styles;

type HeaderProps = {
  viewType: string;
  setViewType: Dispatch<SetStateAction<"table" | "list">>;
};

const Header: FC<HeaderProps> = ({ viewType, setViewType }) => {
  const handleChange = (e: RadioChangeEvent) => {
    setViewType(e.target.value);
  };

  return (
    <Card className={container}>
      <h1 className={title}>1337 fellows</h1>
      <div className={controlsWrap}>
        <Radio.Group value={viewType} onChange={handleChange}>
          <Radio.Button value="table">Table</Radio.Button>
          <Radio.Button value="list">List</Radio.Button>
        </Radio.Group>
      </div>
    </Card>
  );
};

export default Header;
