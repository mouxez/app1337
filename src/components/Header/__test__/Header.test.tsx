import { render } from "@testing-library/react";
import Header from "../Header";

describe("components/Header", () => {
  it("renders correctly", async () => {
    const onClick = jest.fn();
    const { asFragment } = render(<Header viewType="table" setViewType={onClick} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
