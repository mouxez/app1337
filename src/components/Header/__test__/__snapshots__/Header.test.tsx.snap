// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Header renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container container"
  >
    <h1
      class="title"
    >
      1337 fellows
    </h1>
    <div
      class="controlsWrap"
    >
      <div
        class="ant-radio-group ant-radio-group-outline"
      >
        <label
          class="ant-radio-button-wrapper ant-radio-button-wrapper-checked"
        >
          <span
            class="ant-radio-button ant-radio-button-checked"
          >
            <input
              checked=""
              class="ant-radio-button-input"
              type="radio"
              value="table"
            />
            <span
              class="ant-radio-button-inner"
            />
          </span>
          <span>
            Table
          </span>
        </label>
        <label
          class="ant-radio-button-wrapper"
        >
          <span
            class="ant-radio-button"
          >
            <input
              class="ant-radio-button-input"
              type="radio"
              value="list"
            />
            <span
              class="ant-radio-button-inner"
            />
          </span>
          <span>
            List
          </span>
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
